### PR TITLE
Redesign explorer to just do good path handling (Closes #30)

### LINKED ISSUE #30: Design of H5Explorer
Looking at the remaining backend tickets, as well as what we have so far, I'm realizing that H5Explorer is basically just becoming a really light wrapper on top of the existing h5py File object, with the one extra feature being that it remember the working directory and handle relative and absolute file paths.

I'm thinking it would probably make sense to just implement the `__getitem__` function for H5Explorer (and maybe some extra functions for enforcing that something is a Group or Dataset) and resolve the paths and return you the File object with the appropriate path applied. Then, it's still just a 1-line call from the CLI class, but we won't have to update H5Explorer every time we want to implement a new function.

@ksunden What do you think?

### PR DESCRIPTION
**Pull request by lphk92**

### CODE REVIEW

File: src/cli.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 from cmd2 import Cmd, options, make_option
-from . import h5_wrapper
+import explorer
```
Comment (ksunden) [inline src/cli.py]: This needs to be a relative import for execution outside of the src directory (e.g. from the distutils generated entrypoint)

File: src/cli.py [hunk 2]
```diff
@@ -29,23 +30,23 @@ def postcmd(self, stop, line):
     def do_source(self, args):
         Cmd.do_load(self, args)
 
+    @options([
+        make_option('-m', '--mode', type=str, default='r', help='File mode ([r]ead, [w]rite, [a]ppend)')
+    ])
     def do_load(self, args, opts=None):
-        print("Loading " + args)
-        self.explorer = h5_wrapper.H5Explorer(args)
+        self.explorer = new_explorer.NewH5Explorer.from_file(args[0], mode=opts.mode)
```
Comment (ksunden) [inline src/cli.py]: I am getting a NameError new_explorer is not defined here

File: src/explorer.py [hunk 3]
```diff
@@ -2,15 +2,19 @@
...
+        self.__filename = h5_obj.file.filename
+        self.__file = h5_obj
+        self.__working_dir = h5_obj.name
 
         self.__dir_queue = list()
 
+    @staticmethod
+    def from_file(filename, mode="r"):
+        h5_file = h5py.File(filename, mode)
+        return NewH5Explorer(h5_file)
```
Comment (ksunden) [inline src/explorer.py]: NewH5Explorer not defined

File: src/explorer.py [hunk 4]
```diff
@@ -48,37 +52,34 @@ def change_dir(self, new_dir="/"):
...
-    def list_datasets(self, target_dir=None):
-        target_dir = self.__check_group(target_dir)
-
-        return [k for k, v in self.__file[target_dir].items()
-                if isinstance(v, h5py.Dataset)]
+    def __getitem__(self, path):
+        result = self.__file[self.__get_absolute_path(path)]
+        if isinstance(result, h5py.Dataset):
+            return result
+        return NewH5Explorer(result)
```
Comment (ksunden) [inline src/explorer.py]: drop the New

**Review by ksunden — CHANGES_REQUESTED** [review]
Is there any functionality that is known to be lost in this transition (I don't see any, just want to make sure)

**Comment by ksunden** [comment]
@lphk92 I fixed all the errors I found, but I do want to give you the opportunity to check my changes/ensure I didn't miss anything that my changes remind you of.

Once I hear from you, I am ready to merge this

File: src/cli.py (hunk 1 above)
Comment (ksunden) [reply src/cli.py]: I am noticing that it fails running directly from src with the relative import, but it fails from `h5cli` without it, and that is, in my opinion, how it should be accessed in the future

**Comment by lphk92** [comment]
@ksunden How are you running this locally? When I try to run it with the relative import in there I get the following error:

```
Traceback (most recent call last):
  File "./src/cli.py", line 3, in <module>
    from . import explorer
SystemError: Parent module '' not loaded, cannot perform relative import
```

**Comment by lphk92** [comment]
This answer on StackOverflow provides a lot of useful context that to me makes a compelling argument as to why it might make more sense to structure the project such that we have an absolute import there:

https://stackoverflow.com/questions/16981921/relative-imports-in-python-3/28154841#28154841

**Comment by lphk92** [comment]
Just put up a patch that creates a separate script for running locally so relative import works

**Review by ksunden — APPROVED** [review]

**Pull request merged**